js/redot.js

```diff
@@ -0,0 +1,430 @@
+        const datetime = this.iso8601 (timestamp);
+        const symbol = this.safeSymbol (undefined, market);
+        const side = this.safeString (trade, 'side');
+        const price = this.safeNumber (trade, 'price');
+        const amount = this.safeNumber (trade, 'qty');
+        return this.safeTrade ({
+            'info': trade,
+            'id': id,
+            'timestamp': timestamp,
+            'datetime': datetime,
+            'symbol': symbol,
+            'order': id,
+            'type': undefined,
+            'side': side,
+            'takerOrMaker': undefined,
+            'price': price,
+            'amount': amount,
+            'cost': undefined,
+            'fee': undefined,
+        });
+    }
+
+    parseTicker (ticker, market = undefined) {
+        //
+        // {
+        //     "lastTradeId":7219332,
+        //     "price":0.068708,
+        //     "qty":0.0046,
+        //     "bidPrice":0.068663,
+        //     "askPrice":0.068678,
+        //     "bidQty":0.4254,
+        //     "askQty":0.0506,
+        //     "volumeUsd":315646.03,
+        //     "volume":130.5272,
+        //     "time":1642974178845710
+        // }
+        //
+        const symbol = this.safeSymbol (undefined, market);
+        const last = this.safeString (ticker, 'price');
+        const open = this.safeString (ticker, 'openPrice');
+        const high = this.safeString (ticker, 'highPrice');
+        const low = this.safeString (ticker, 'lowPrice');
+        const baseVolume = this.safeString (ticker, 'volume');
+        const bid = this.safeString (ticker, 'bidPrice');
+        const ask = this.safeString (ticker, 'askPrice');
+        let timestamp = Precise.stringDiv (this.safeString (ticker, 'time'), '1000');
+        timestamp = parseInt (parseFloat (timestamp));
+        return this.safeTicker ({
+            'symbol': symbol,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': high,
+            'low': low,
+            'bid': bid,
+            'bidVolume': undefined,
+            'ask': ask,
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': open,
+            'close': last,
+            'last': last,
+            'previousClose': undefined,
+            'change': undefined,
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': baseVolume,
+            'quoteVolume': undefined,
+            'info': ticker,
+        }, market, false);
+    }
+
+    parseOHLCV (ohlcv, market = undefined) {
+        //
+        //  {
+        //       "time":1642973286229265,
+        //       "low":0.068584,
+        //       "high":0.068584,
+        //       "open":0.068584,
+        //       "close":0.068584,
+        //       "volume":0.0338
+        //  },
+        //
+        return [
+            parseInt (parseFloat (Precise.stringDiv (this.safeString (ohlcv, 'time'), '1000'))),
```

Comment:
we should use `this.safeInteger` here because it is already in milliseconds

Reply:
@frosty00 Not sure if I understood, the response's timestamp is in Microseconds right? So needs to be converted to milliseconds

Reply:
yep, i think we can still avoid using `Precise` and just do 

```
const microseconds = this.safeInteger
let milliseconds = undefined
if (microseconds !== undefined) {
   milliseconds = parseInt (microseconds / 1000)
}
```

Reply:
@carlosmiei in this case, since we're dealing with values under 64 bit, we can simply do `this.safeIntegerProduct (ohlcv, 'time', 0.001)`.

Reply:
The same is true in other similar cases throughout the file.

Reply:
@kroitor didn't know this method but will use